src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx

```diff
@@ -111,43 +136,99 @@ class MonitorForm extends Component {
                   label={t('Type')}
                   disabled={!hasAccess}
-                  choices={[['cron_job', 'Cron Job']]}
+                  choices={MONITOR_TYPES}
                   required
                 />
                 <Observer>
                   {() => {
-                    return (
-                      this.form.getValue('type') === 'cron_job' && (
-                        <SelectField
-                          name="config.schedule_type"
-                          label={t('Schedule Type')}
-                          disabled={!hasAccess}
-                          choices={[['crontab', 'Crontab']]}
-                          required
-                        />
-                      )
-                    );
+                    switch (this.form.getValue('type')) {
+                      case 'cron_job':
+                        return (
+                          <div>
+                            <NumberField
+                              name="config.max_runtime"
+                              label={t('Max Runtime')}
+                              disabled={!hasAccess}
+                              help={t(
+                                "The maximum runtime (in minutes) a check-in is allowed before it's marked as a failure."
+                              )}
+                              placeholder="e.g. 30"
+                            />
+                            <SelectField
+                              name="config.schedule_type"
+                              label={t('Schedule Type')}
+                              disabled={!hasAccess}
+                              choices={SCHEDULE_TYPES}
+                              required
+                            />
+                          </div>
+                        );
+                      default:
+                        return null;
+                    }
                   }}
                 </Observer>
                 <Observer>
                   {() => {
-                    return (
-                      this.form.getValue('config.schedule_type') === 'crontab' && (
-                        <TextField
-                          name="config.schedule"
-                          label={t('Schedule')}
-                          disabled={!hasAccess}
-                          placeholder="*/5 * * *"
-                          required
-                          help={tct(
-                            'Changes to the schedule will apply on the next check-in. See [link:Wikipedia] for crontab syntax.',
-                            {
-                              link: <a href="https://en.wikipedia.org/wiki/Cron" />,
-                            }
-                          )}
-                        />
-                      )
-                    );
+                    switch (this.form.getValue('config.schedule_type')) {
+                      case 'crontab':
+                        return (
+                          <div>
```

Comment:
Pedantic but these can probably be `<React.Fragment>`s right?

Reply:
Oh right forgot that existed again